### PR TITLE
[14.0][FIX] account_invoice_pricelist: Make tests resilient

### PR DESCRIPTION
**Pull request by pedrobaeza**
- Depending on the installed set of modules, the company currency may be USD or EUR. If the second case, these tests will fail, so we make sure that the company currency is USD for our tests, doing the change by SQL, as there's a Python constraint that prevents it.
- On contrary, the main pricelist doesn't change its currency, so we can have an incompatible currency.
- The onchange partner test should assert fixed data.

@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
FYI, this weird company's currency switch has been finally solved in odoo/odoo#107113 for v17.

File: account_invoice_pricelist/tests/test_account_move_pricelist.py [hunk 1]
```diff
@@ -27,6 +27,12 @@ def setUpClass(cls):
         cls.journal_sale = cls.env["account.journal"].create(
             {"name": "Test sale journal", "type": "sale", "code": "TEST_SJ"}
         )
+        # Make sure the currency of the company is USD, as this not always happens
```
Comment (legalsylvain) [inline account_invoice_pricelist/tests/test_account_move_pricelist.py:30]: If I understand correctly your point, regarding V17.

```suggestion
        # Make sure the currency of the company is USD, as this not always happens
        # Remove this SQL request when porting this module in V17.
        # See : https://github.com/odoo/odoo/pull/107113 
```

Comment (pedrobaeza) [reply account_invoice_pricelist/tests/test_account_move_pricelist.py:30]: Added, @legalsylvain

**Review by ernestotejeda — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**